Tasks/UseDotNetV2/versioninstaller.ts

```diff
@@ -43,7 +44,9 @@ export class VersionInstaller {
             // Extract
             console.log(tl.loc("ExtractingPackage", downloadPath));
             try {
-                var extPath = tl.osType().match(/^Win/) ? await toolLib.extractZip(downloadPath) : await toolLib.extractTar(downloadPath);
+                let tempDirectory = tl.getVariable('Agent.TempDirectory');
+                var destination = path.join( tempDirectory, this._tinyGuid());
+                var extPath = tl.osType().match(/^Win/) ? await toolLib.extractZip(downloadPath, destination) : await toolLib.extractTar(downloadPath);
```

Comment:
Do we also need to update for linux os ?

Reply:
Can you verify that the same long path works on linux, if not then we should update for linux as well

Reply:
verified, its working fine on linux agent